tests/EdFi.Ods.Api.IntegrationTests/EducationOrganizationTestDataBuilder.cs

```diff
@@ -228,5 +232,86 @@ public int Execute()
             _sql.Clear();
             return result;
         }
+
+        public EducationOrganizationTestDataBuilder AddStudent(string newGuidId)
+        {
+            _sql.AppendLine(
+                $@"INSERT INTO edfi.Student (
+                    FirstName,
+                    LastSurname,
+                    BirthDate,
+                    StudentUniqueId)
+                VALUES (
+                    '{newGuidId}',
+                    '{newGuidId}',
+                    '{DateTime.UtcNow.Date}',
+                    '{newGuidId}');"
+            );
+
+            return this;
+        }
+
+        public EducationOrganizationTestDataBuilder AddParent(string newGuidId)
+        {
+            _sql.AppendLine(
+                $@"INSERT INTO edfi.Parent (
+                    FirstName,
+                    LastSurname,
+                    ParentUniqueId)
+                VALUES(
+                    '{newGuidId}',
+                    '{newGuidId}',
+                    '{newGuidId}');"
+            );
+
+            return this;
+        }
+
+        public EducationOrganizationTestDataBuilder GetStudentUSI(string studentUniqueId)
+        {
+            _sql.AppendLine(
+                $@"SELECT StudentUSI FROM edfi.Student
+                WHERE StudentUniqueId = '{studentUniqueId}';");
+
+            return this;
+        }
+
+        public EducationOrganizationTestDataBuilder GetParentUSI(string parentUniqueId)
+        {
+            _sql.AppendLine(
+                $@"SELECT ParentUSI FROM edfi.Parent
+                WHERE ParentUniqueId = '{parentUniqueId}'; ");
+                
+            return this;
+        }
+
+        public int ExecuteScalar()
+        {
+            using var command = Connection.CreateCommand();
+            command.CommandText = _sql.ToString();
+            var result = Convert.ToInt32(command.ExecuteScalar());
+
+            _sql.Clear();
+            return result;
+        }
+
+         public EducationOrganizationTestDataBuilder AddStudentSchoolAssociation(int schoolId, int studentUSI, int gradeLevelDescriptorId)
```

Comment:
Do we care about the grade level?  If you look at the pattern of the other builder methods, if there's a test descriptor that is required but isn't important to the identity then we usually grab the first one in the initialize and just consume it internally.  What's probably more important here is handling EntryDate as a provided parameter, since you're going to want to do some tests where you have the same student enrolled at the same school with different entry dates, since its part of the key.  I'd recommend providing it as a parameter but providing a default on it.

Reply:
I updated as per your comments